src/vs/base/browser/ui/splitview/splitview.ts

```diff
@@ -197,6 +198,15 @@ export namespace Sizing {
 	export function Invisible(cachedVisibleSize: number): InvisibleSizing { return { type: 'invisible', cachedVisibleSize }; }
 }
 
+export interface ISplitViewDescriptor {
+	size: number;
+	views: {
+		visible?: boolean;
+		size: number;
+		view: IView;
```

Comment:
@sbatten I might be wrong but I believe this won't work in the big picture. Remember that a gridview is a tree of splitviews. In order to fully deserialize a whole gridview in one step, you're very likely going to have to use the deserializer pattern in which a `fromJSON` method is called, even down in the splitview.

Reply:
the grid view deserializes views using a depth first search so this shouldn't be an issue